File: pset1/greedy.c
// hello my name is mostafa radwan and this is CS50 :))

#include <cs50.h>
#include <stdio.h>
#include <math.h>
int main (void)

{
    
    printf("O hai! How much change is owed?\n");
   float a =GetFloat();// user value
  int b=round(a*100); 
    int y=0 ;  //  counter
    
    if (b>0)    // accepts positve values
    {
    while (b>0)            
    {
                       // quarters checkers
        if (b>=25)
        {
    b=b-25 ;
    y++;  //   printf("%i %i\n" ,y,b);
}
    
    else if  (25>b && b>=10)
        {
    b=b-10 ;
    y++ ; // printf("%i %f\n" ,y,b);
     }
    
    else if(10>b && b>=5){
    b=b-5;
    y++;
   // printf(" check %i \n",y);
  }
   
    else if  (5>b && b>=1){
    b=b-1;
   y++;
        
    }

    }
        printf("%d\n" ,y);
}
else {
    printf ("retry");
       a =GetFloat();// user value
       b=round(a*100); 
}
}
    

File: pset66/greedy.py
import cs50
def main():
    
 print ("O hai! How much change is owed? ")
 x = float(input())
 x = round(x*100,0)
 print(x)
 b=0
 while (x>0):
  if (x>=25):
   x=x-25
   b+=1
  if (25>x and x>=10): 
   x=x-10
   b+=1
   
  if (10>x and x>=5):
    x=x-5
    b+=1
  if  (5>x and x>=1):
    x=x-1
    b+=1
 print (b)    
    

if __name__ == "__main__":
    main()
